pytripgui/leftmenu.py

```diff
@@ -328,7 +321,38 @@ def plan_field_added(self, msg):
         self.AppendItem(fields, field.get_name(), data=data)
 
     def plan_run_trip(self, evt):
+        """
+        Callback for TRiP98 Execution.
+        This will set the last global parameters and then execute TRiP for the attached plan.
+        """
         plan = self.GetItemData(self.selected_item).GetData()
+
+        # Load global parameters from settings file and attach them to this plan.
+        st = Settings()
+        plan.set_remote_state(st.load('trip98.choice.remote') == '1')  # remote if set to '1'
+        plan.set_working_dir(st.load('trip98.s.wdir'))
+        plan.set_server(st.load('trip98.s.server'))
+        plan.set_username(st.load('trip98.s.username'))
+        plan.set_password(st.load('trip98.s.password'))
+
+        # tell what configuration files to be used, depending on the
+        # projectile / rifi configuration
+
+        # if not specified, use protons without RiFi
+        if not hasattr(plan, "_projectile"):
+            plan._projectile = 0
+        if not hasattr(plan, "_rifi"):
+            plan._rifi = 0
+
+        _dion = ('z1', 'z2', 'z6', 'z8', 'z10')
+        _drifi = ("rifi0", "rifi3")
+
+        _suffix = '{:s}{:s}'.format(_dion[plan._projectile], _drifi[plan._rifi])
+
+        plan.set_ddd_folder(st.load('trip98.ddd.{:s}'.format(_suffix)))
+        plan.set_spc_folder(st.load('trip98.spc.{:s}'.format(_suffix)))
+        plan.set_sis_file(st.load('trip98.sis.{:s}'.format(_suffix)))
+
```

Comment:
The weird part is: multiple ions seem to be supported in the Field and in the former "Advanced Dose" tab for the plan object. However: only one set of ddd and spc files can be loaded.